arangod/Aql/OptimizerRule.h

```diff
@@ -299,6 +299,10 @@ struct OptimizerRule {
     // parallelizes execution in coordinator-sided GatherNodes
     parallelizeGatherRule,
 
+    // no document materialization for arango search view
+    // if stored values contain all fields
+    noDocumentMaterializationArangoSearchRule,
```

Comment:
let's try to combine `noDocumentMaterializationArangoSearchRule` and `handleViewRule`,
and introduce options to control `noDocumentMaterializationArangoSearchRule` behaviour